src/status_im/data_store/contacts.cljs

```diff
@@ -34,6 +35,33 @@
 (defn- get-contact-by-id [public-key realm]
   (core/single (core/get-by-field realm :contact :public-key public-key)))
 
+(defn get-blocked-user-data
+  [public-key]
+  {:messages (core/all-clj (core/get-by-field @core/account-realm
+                                              :message :from public-key)
+                           :message)
+   :chat (core/single-clj (core/get-by-field @core/account-realm
+                                             :chat :chat-id public-key)
+                          :chat)})
+
+(re-frame/reg-cofx
+ :data-store/get-blocked-user-data
+ (fn [cofx _]
+   (assoc cofx :get-blocked-user-data get-blocked-user-data)))
+
+(defn block-user-tx
+  "Returns tx function for deleting user messages"
+  [{:keys [public-key] :as contact}]
+  (fn [realm]
+    (core/create realm :contact contact true)
+    (let [chat (core/get-by-field realm :chat :chat-id public-key)
+          messages (core/get-by-field realm :message :from public-key)
+          statuses (core/get-by-field realm :user-status :public-key public-key)]
```

Comment:
We can also try to fetch statuses by primary key because we do know them here, it might be faster. Depends on quantities